### ISSUE TITLE
Ephys Link automation settings

### ISSUE THREAD

**Issue by kjy5** [VirtualBrainLab/Pinpoint#290]
Add new tab in settings for automation preferences for Ephys Link. Should clearly differentiate settings for when inside the brain vs outside.

- Settle timing function
- Outside brain drive speeds: Default @ 100 um/s; limit @ 1000 um/s
- Inside brain drive speeds: Default @ 2 + 1 um/s per 1000 um of depth; limit to 100 um/s. Return to target at 2 um/s
- Exit brain speeds: default @ 10 um/s, drive past dura for 100 um at 25 um/s, go rest of the way at outside brain driving speeds

**Comment by kjy5** [comment]
- Show time estimate for drive speed
- 3-speed buttons: safe, fast, testing (show warning)

**Comment by dbirman** [comment]
Actually the speed settings should probably become an independent feature separated from both ProbeController and from the EphysLink code. I was thinking of changing how it works to be that if you press ctrl it slows down one speed and shift speeds up one speed (or on the controller two adjacent buttons, not sure which yet). We'd show the current speed somewhere in the viewport, and then this speed setting propagates out to all ProbeController objects. 

Thoughts before I implement that?

**Comment by kjy5** [comment]
<img width="318" alt="image" src="https://github.com/VirtualBrainLab/Pinpoint/assets/82800265/8b952e70-1285-469c-97db-3bd40fc3c5c7">

UI looks something like this right now where the user would click one of these buttons to set the speed and drive.

I'm hesitant to have a universal speed used everywhere because while somebody may want to drive at a slower speed, they may want to rearrange probes in the scene faster. If it were all the same that might get confusing. (or maybe I'm misunderstanding the proposal)

Basically, I think it makes sense for there to be these speed buttons somebody could press before they drive that way they can decide what they want to do per probe

**Comment by dbirman** [comment]
Hmm, good point that automation probably needs a different speed from keyboard/controller movements. Let's stick with the separated plan for now!
